### PR TITLE
Podman and podman dependencies

### PR DESCRIPTION
**Pull request by knusbaum**
This is broken up into 3 commits for the 3 packages required by podman.

### CODE REVIEW

**Comment by CameronNemo** [comment]
Do these really need to override so much of the go build style?

Why are the cni configuration files installed with libpod rather than cni?

Use the vmkdir and vinstall commands wherever possible (see the manual for usage instructions).

Why not name the package libpod to match the repo name?

**Comment by knusbaum** [comment]
Makefile patch is here:
https://github.com/kubernetes-sigs/cri-o/pull/1933

File: srcpkgs/cni-plugins/template [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+distfiles="https://github.com/containernetworking/plugins/archive/v${version}.tar.gz"
+checksum=d47f2bda2bfd6699843a98d91a2e5c40aa58eacd3f2393c7696c158c410aad3c
+
+do_build() {
+	./build.sh
+}
+
+do_install() {
+	vmkdir /usr/libexec/cni
+	cp bin/* ${DESTDIR}/usr/libexec/cni
```
Comment (maxice8) [inline srcpkgs/cni-plugins/template]: I might be wrong but i think vcopy can be used here

`vcopy bin/* usr/libexec/cni`

File: srcpkgs/cni-plugins/template [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+	vmkdir /etc/cni/net.d
+	vinstall ${FILESDIR}/10-default.conf 664 /etc/cni/net.d/
```
Comment (maxice8) [inline srcpkgs/cni-plugins/template]: if you use vinstall then you don't need to use vmkdir.

Comment (maxice8) [reply srcpkgs/cni-plugins/template]: Explanation: vinstall uses `install -D` which creates all leading directories.

**Comment by knusbaum** [comment]
Requested changes addressed including quoting globing.

**Comment by CameronNemo** [comment]
Thanks for making those changes. This is looking quite good. Some notes after building and inspecting the packages:

* in the cni-plugins template, add `conf_files="/etc/cni/net.d/*.conf"` below the checksum line
* in the podman template, add `conf_files="/etc/containers/registries.conf"` below the checksum line
* in the cri-o and podman templates, add `go-md2man` to the hostmakedepends
* in the podman template, there are systemd units and tmpfiles snippets which should probably be removed
* the podman template should define `pycompile_module`, likely `pycompile_version`, and possibly `pycompile_dirs`

**Comment by knusbaum** [comment]
Thanks, sounds good. 

I only have questions about this:
> in the cri-o and podman templates, add go-md2man to the hostmakedepends

I'm not sure we want this. The tools make target for these packages installs go-md2man if it's not in `$GOPATH/bin` regardless, so I don't think we gain anything by making it a host make dependency.

I'd love to not have the build install this, but that's an upstream issue.

There is the option that I can break up install.tools and manually run all the other targets *except* go-md2man, but again, not sure we really gain anything through this. I think it will only make the template harder to maintain.

**Comment by knusbaum** [comment]
Also realized I was failing to build and package the python modules, so I've fixed that.

**Comment by jcgruenhage** [comment]
@CameronNemo That would be great :)

**Comment by knusbaum** [comment]
@CameronNemo I saw that you put up and merged an MR for cni-plugins, so I thought this MR was not wanted. I can reopen if you want.

**Comment by knusbaum** [comment]
Ahh, nevermind. I see #5616

**Pull request closed without merging**